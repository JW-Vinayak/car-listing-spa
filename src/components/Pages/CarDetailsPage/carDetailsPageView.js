import React from "react";
import CarDetails from "../../CarDetails/carDetailsView";
import { WithHeaderAndFooter } from "../../../utils/composition";


const CarDetailsPage = props => {
  return (
    <WithHeaderAndFooter>
      <CarDetails {...props} />
    </WithHeaderAndFooter>
  );
};

export default CarDetailsPage;
