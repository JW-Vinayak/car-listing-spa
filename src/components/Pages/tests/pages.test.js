import React from "react";
import CarDetailsPage from "../CarDetailsPage/carDetailsPageView";
import PageNouFoundPage from "../NotFoundPage/notFoundPageView";
import renderer from "react-test-renderer";

let match = {
  params: {
    stockNumber: 1234
  }
};

describe("CarDetailsPage component", () => {
  it("shallow renders without crashing", () => {
    const tree = renderer.create(<CarDetailsPage match={match} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("PageNouFoundPage component", () => {
  it("shallow renders without crashing", () => {
    const tree = renderer.create(<PageNouFoundPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
