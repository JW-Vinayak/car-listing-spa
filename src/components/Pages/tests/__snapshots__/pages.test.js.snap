// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarDetailsPage component shallow renders without crashing 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="header-logo"
    >
      <img
        alt="Company"
        className="logo"
        src="company-logo.png"
      />
    </div>
    <div
      className="header-menu"
    >
      <a
        className="header-link"
        href="##"
      >
        Purchase
      </a>
      <a
        className="header-link"
        href="##"
      >
        My Orders
      </a>
      <a
        className="header-link"
        href="##"
      >
        Sell
      </a>
    </div>
  </header>,
  <div
    className="error-box"
  >
    Data is Loading...
  </div>,
  <footer
    className="footer"
  >
    <div
      className="copyright"
    >
      © AUTO1 Group 2018
    </div>
  </footer>,
]
`;

exports[`PageNouFoundPage component shallow renders without crashing 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="header-logo"
    >
      <img
        alt="Company"
        className="logo"
        src="company-logo.png"
      />
    </div>
    <div
      className="header-menu"
    >
      <a
        className="header-link"
        href="##"
      >
        Purchase
      </a>
      <a
        className="header-link"
        href="##"
      >
        My Orders
      </a>
      <a
        className="header-link"
        href="##"
      >
        Sell
      </a>
    </div>
  </header>,
  <section
    class="page-not-found"
  >
    <img
      alt="Company"
      className="logo"
      src="company-logo.png"
    />
    <header>
      404 - Not Found
    </header>
    <p
      className="text"
    >
      Sorry, the page you are looking for does not exist.
    </p>
    <p
      className="text"
    >
      You can always go back to
       
      <a
        className="link"
        href="/"
      >
        homepage
      </a>
    </p>
  </section>,
  <footer
    className="footer"
  >
    <div
      className="copyright"
    >
      © AUTO1 Group 2018
    </div>
  </footer>,
]
`;
